src/libraries/Common/tests/System/Collections/DictionaryEntry.Tests.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Collections.Tests
+{
+    public class DictionaryEntry_Tests
+    {
+        public static IEnumerable<object?[]> ToString_Inputs()
+        {
+            yield return new object?[] { "key", "value", "[key, value]" };
+            yield return new object?[] { 1, 2, "[1, 2]" };
+            yield return new object?[] { typeof(object), (object?)null, "[System.Object, ]" };
+            yield return new object?[] { "[key, value", "]]", "[[key, value, ]]]" };
+            yield return new object?[] { new DictionaryEntry("key", "key"), new DictionaryEntry("value", "value"), "[[key, key], [value, value]]" };
+            yield return new object?[] { new String('K', 512), new String('V', 1024), $"[{new String('K', 512)}, {new String('V', 1024)}]" };
+        }
```

Comment:
Key can be null as well (even though the nullable annotation suggests it shouldn't be).  It'd be good to add tests for that.  And for empty strings for both key and value.